lib/util/extractSourceMap.js

```diff
@@ -0,0 +1,296 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Ivan Kopeykin @vankop
+*/
+
+"use strict";
+
+const path = require("path");
+const urlUtils = require("url");
+const WebpackError = require("../WebpackError");
+const { isAbsolute, join } = require("./fs");
+
+/** @typedef {import("../NormalModule")} NormalModule */
+/** @typedef {import("./fs").InputFileSystem} InputFileSystem */
+
+// Matches only the last occurrence of sourceMappingURL
+const innerRegex = /\s*[#@]\s*sourceMappingURL\s*=\s*([^\s'"]*)\s*/;
+
+/* eslint-disable prefer-template */
+const sourceMappingURLRegex = RegExp(
+	"(?:" +
+		"/\\*" +
+		"(?:\\s*\r?\n(?://)?)?" +
+		"(?:" +
+		innerRegex.source +
+		")" +
+		"\\s*" +
+		"\\*/" +
+		"|" +
+		"//(?:" +
+		innerRegex.source +
+		")" +
+		")" +
+		"\\s*"
+);
+/* eslint-enable prefer-template */
+
+/**
+ * @param {string} code code
+ * @returns {{replacementString: null|string, sourceMappingURL: null|string}} source mapping url
+ */
+function getSourceMappingURL(code) {
+	const lines = code.split(/^/m);
```

Comment:
`.split()` is very expensive. I'd recommend using repeated calls to `lastIndexOf('\n', lastNewline)`, especially since this regexp only needs to match the last one.

Reply:
We can't use `lastIndexOf` due node.js version support range

Reply:
Really? `String.prototype.lastIndexOf(searchString, position)` is ancient.

Reply:
Oh, yeah, I confused with another

Reply:
@vankop can we make this change?

Reply:
Done.